### DISCUSSION TITLE
BACnet

### DISCUSSION THREAD

**Discussion by nas2011** [toitlang/toit#564] [General]
Just curious if there has been any discussion about building tools to support the use of BACnet protocol? When it comes to industrial building automation it is one of those standards that if you don't support it you are already out of the running. 

https://www.techstreet.com/ashrae/standards/ashrae-135-2020?gateway_code=ashrae&product_id=2191852

Commentary:
     I'm not sure how many millions of BAS points are out there using BACnet but it has to be a huge market. I know where I work we have tons of BACnet points and since only large automation companies like siemens and johnson controls have put in the leg work they charge a fortune to get a simple sensor up and running. If the tools for BACNET support were available, the hardware is already relatively cheap. This would allow capable companies to build in to their existing BAS in an affordable and flexible manner. I really think the numbers are such that the savings from the first 10 would fund the next 100 devices' operating fees and hardware. Also I think that toit could add a great measure of robustness by allowing logging and health checking features in addition to and independent from sensor readings.

**Comment by floitsch** [comment]
Just had a look at BACnet.

It should be a good fit as it works in the embedded space. However, the standard is relatively big and hard to implement. There are companies that make a living by selling their libraries...
Hopefully there will be a package eventually, but at the moment we don't have the resources to implement such a big standard.

**Reply by nas2011** [reply]
That is what I figured but thought I would ask just to be sure. Thanks for the feedback.
